pkg/osversion/osversion_test.go

```diff
@@ -0,0 +1,125 @@
+package osversion_test
+
+import (
+	"fmt"
+	"testing"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/ginkgo/extensions/table"
+	. "github.com/onsi/gomega"
+	"github.com/openshift-psap/special-resource-operator/pkg/osversion"
+)
+
+func TestOsversion(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Osversion Suite")
+}
+
+type testCase struct {
+	rel        string
+	maj        string
+	min        string
+	out0       string
+	out1       string
+	out2       string
+	expectsErr bool
+}
+
+func (tc *testCase) String() string {
+	str := fmt.Sprintf(
+		"%s %s.%s => [%s, %s, %s]",
+		tc.rel,
+		tc.maj,
+		tc.min,
+		tc.out0,
+		tc.out1,
+		tc.out2)
+
+	if tc.expectsErr {
+		str += " (error expected)"
+	}
+
+	return str
+}
+
+var _ = Describe("OSVersion", func() {
+	Context("RenderOperatingSystem", func() {
+		cases := []*testCase{
```

Comment:
looks good, but i think you can write it a little bit more clear without DescribeTable and with it in the loop. Please see:
https://github.com/openshift/assisted-service/blob/1b137112f1a4b5b7944748c6a9353e41573f2c7c/internal/cluster/cluster_test.go#L696

WDYT?

Reply:
Sure, that's actually closer to what I'm used to writing - I thought `DescribeTable` was the best practice with Ginkgo.

Reply:
Make sure you don't forget this part https://github.com/openshift/assisted-service/blob/1b137112f1a4b5b7944748c6a9353e41573f2c7c/internal/cluster/cluster_test.go#L742 as it is crucial, without it only the last test in the table list will actually run (all other will be overridden by the last)

Reply:
Isnt `DescribeTable` designed to mimick table testing in `testing` framework? It does not look so obvious/easy to read for me to take things from `testing` and then `It`-ing each test case. Also, the name aliasing to avoid taking the bad references is prone to error too.
Pardon my ignorance in Ginkgo, but why revert to `testing` instead of using `DescribeTable`?

Reply:
i don't think it is a revert to testing , Ginko also has It. It is just a suggestion of a more readable code (from my point a view).  just a suggestion

Reply:
Oh dont get me wrong, I was just curious if there was some common practice/recommendation about `DescribeTable`, as I have already included that in a test. Thanks!

Reply:
I think I would stick with the Ginkgo way of doing things here. It is a bit unfortunate indeed that we need to make 2 slices (test cases and then Ginkgo entries), but I still prefer this over creating the `It`s manually, which is more error-prone IMHO. Hopefully this can be avoided in future table tests that contain a `name` field and directly replaced with `Entry()`.